### ISSUE TITLE
Index FOLIO course reserves data

### ISSUE THREAD

**Issue by cbeer** [sul-dlss/searchworks_traject_indexer#667]
We punted on all the CREZ data indexing for FOLIO. Unclear where we're getting that reserves data in FOLIO, but there's probably work for us to do there.

**Comment by thatbudakguy** [comment]
@saseestone is going to create some fixture data for us in FOLIO and make sure it's associated with items for testing.

**Comment by saseestone** [comment]
1 course created

Course code: AFRICAAM-105-01
Course name: Black Matters: Introduction to Black Studies

Cross-listed with course code: AFRICAAM-269-01 

It's got two items associated with it:
item barcode: 36105021019562
item barcode: 36105021837179

**Comment by saseestone** [comment]
Another fixture course:

Course code: AFRICAAM-158-01
Course name: Rebellion: Black Resistance in the Caribbean
Item ID associated with course: 36105212524685

**Comment by cbeer** [comment]
Thanks. We're able to pull back data from the course reserves API.

We're unsure how FOLIO handles the current location + loan period when an item is actively on reserve (or if we continue to need our Symphony data rewriting hacks)

**Comment by saseestone** [comment]
Ok one fun fact - if an item is associated with a course, it's "on reserve". FOLIO doesn't have any way (currently) to prepare Item records ahead of the quarter or stage them in some way.  Unlike Symphony, there aren't reserve "status" like "prepare" or "dormant".  

So a course record has either:

A. Items associated with the course. and that therefore all the items are "on reserve"
OR
B. No items associated with the course, and therefore no items are on reserve (and we shouldn't display anything about the Course in SearchWorks.

**Comment by saseestone** [comment]
@cbeer I think a step was missed in the "How to create Course records" demo. (Or I just missed seeing it.) 

It turns out that staff have to assign a temporary loan period AFTER they have associated the item with the Course.  

When I associated the first few items with courses, I did NOT update the item to use a temporary loan period.  So when I tried to checkout an item (36105021837179 - Title: Lions, black skins...) it applied the one year loan. That made me seriously scratch my head about what was going on.

But after going back to the Courses app, and selecting to edit the item in the FOLIO ui, I can see a field to input a "Temporary loan type".  I edited 36105022956788 (Title: Wounds of the spirit...) to use a "2-hour loan type".  After that update, I checked out the item and got a due date of 8/16/22 at 2:00pm. (Not 2 hours, but I suspect that it's because I'm close to closing time and the due date is rolling forward. We do this in Symphony, so I'm assuming they are doing this in FOLIO too.)

Does that help?

**Comment by saseestone** [comment]
Successfully added a "Fast Add" item to course id = AFRICAAM-105-01. Item barcode: GRPC0005C01

**Comment by thatbudakguy** [comment]
We're retrieving some of this info from the database when indexing now, but not using it yet:
https://github.com/sul-dlss/searchworks_traject_indexer/blob/4d6a9d33e674344f9e0e6b20263e5c8b2709ae22/lib/traject/readers/folio_postgres_reader.rb#L68-L79

**Comment by thatbudakguy** [comment]
Old PR https://github.com/sul-dlss/searchworks_traject_indexer/pull/686 might've been created prior to switching to database-based indexing strategy, so no longer relevant — but it might include some useful snippets

**Comment by thatbudakguy** [comment]
You can find items on reserve by clicking on the "course reserves" box on the SearchWorks homepage, e.g.: https://searchworks.stanford.edu/view/10765880

Check how the "reserves" dropdown appears on the result list, and how the "reserves" box appears on the show page:

![Image](https://user-images.githubusercontent.com/4924494/233708550-33dcc38e-77eb-4f90-90df-08c5b472237e.png)

**Comment by saseestone** [comment]
Lots of courses in folio-test now to work with. Notable examples:

Cross-listed courses (e.g. classes with the same item list, but taught under multiple course numbers)
- Course Name: Superhero Theory. Course IDs: ARTHIST-120-01, ARTHIST-320-01, AMSTUD-120B-01, FILMEDIA-120-01, FILMEDIA-320-01
- Course Name: The Spirit of Democracy. Course IDs: COLLEGE-110-07, COLLEGE-110-06

Courses with multiple instructors:
- LAW-6017-01
- EDUC-147-01

**Comment by thatbudakguy** [comment]
copying here some questions from the earlier PR:

- [ ] do we still need to rewrite `item_display` to update the current location?

- [ ] do we still need to update `building_facet` to set the library for the reserves desk location, or does the effective location have that info too?

for courses that have a location set (not all do), e.g. [The Romans](https://folio-test.stanford.edu/cr/courses/a9e64aa9-6cb2-4ee7-8827-ca7d2492a979?filters=locations.4a911835-3488-4d26-9293-bdc625c9afce&sort=name), it appears that that location is used to determine the effective location for items on reserve for that course (e.g. [A Brief History of the Romans](https://folio-test.stanford.edu/inventory/view/52d29656-28a5-52b9-a5c3-8b108f4f8ef4/928579d0-1afd-5099-9af4-23e877c96210/305351dd-2102-5e72-a6ac-0efb24876c19), which normally resides in green stacks but has an effective location of green reserves).

- [ ] can we trust the FOLIO department object name for display?

 i think we confirmed that we don't need to index it at all — in the earlier PR we wrote that `crez_dept_facet` and `crez_desk_facet` were unused (in a comment); and they don't appear to be referenced in [SW's `CourseReserves` concern](https://github.com/sul-dlss/SearchWorks/blob/master/app/models/concerns/course_reserves.rb#L1).
